datajoint/expression.py

```diff
@@ -537,6 +591,14 @@ def tail(self, limit=25, **fetch_kwargs):
 
     def __len__(self):
         """:return: number of elements in the result set e.g. ``len(q1)``."""
+        subquery = None
+        if hasattr(self, "top_restriction") and self.top_restriction:
+            subquery = "(SELECT DISTINCT {fields} FROM {from_}{where}{sorting}) AS subquery".format(
```

Comment:
why `DISTINCT`?

Reply:
I wasn't sure how close the subquery should be to the original outer query. Is it not necessary?